### PR TITLE
Add WindowHandler

### PR DESCRIPTION
**Pull request by muscovite**
Extract logic of generating and tracking upcoming characters out of Attempter

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -63,44 +64,65 @@ fn update_stats(start: Instant, typed: usize, attempts: usize) {
     );
 }
 
-struct Attempter {
-    stdout: RawTerminal<io::Stdout>,
+struct WindowHandler {
     practice_set: PracticeSet,
-    chunk: VecDeque<String>,
+    chunk_vec: VecDeque<VecDeque<char>>,
```
Comment (nrxus) [inline src/main.rs]: let's call it words

File: src/main.rs [hunk 2]
```diff
@@ -63,44 +64,65 @@ fn update_stats(start: Instant, typed: usize, attempts: usize) {
...
+    // Return next character to type
+    fn next_char(&mut self) -> char {
+        match self.chunk_vec[0].pop_front() {
             Some(c) => c,
             None => {
-                self.chunk.pop_front();
-                self.chunk
-                    .push_back(self.practice_set.choose().chars().rev().collect());
+                self.chunk_vec.pop_front();
+                let next_word: VecDeque<_> = self.practice_set.choose().chars().collect();
```
Comment (nrxus) [inline src/main.rs]: is the type annotation necessary? :thinking:

Comment (muscovite) [reply src/main.rs]: 🙄

File: src/main.rs [hunk 3]
```diff
@@ -63,44 +64,65 @@ fn update_stats(start: Instant, typed: usize, attempts: usize) {
     );
 }
 
-struct Attempter {
-    stdout: RawTerminal<io::Stdout>,
+struct WindowHandler {
```
Comment (nrxus) [inline src/main.rs]: WordHandler maybe?

**Review by nrxus — APPROVED** [review]
some nits: :+1:

**Pull request merged**